src/node_watcher.js

```diff
@@ -317,6 +317,11 @@ NodeWatcher.prototype.processChange = function(dir, event, file) {
 
 NodeWatcher.prototype.emitEvent = function(type, file, stat) {
   var key = type + '-' + file;
+  var addKey = ADD_EVENT + '-' + file;
+  if (type === CHANGE_EVENT && this.changeTimers[addKey]) {
```

Comment:
I don't think change events on directory should make it to `emitEvent`. We should filter them somewhere up the stack.